great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -195,61 +202,89 @@ def _generate_batch_spec_parameters_from_batch_definition(
+            else:
+                return PathBatchSpec(batch_spec)
 
     @staticmethod
     def _get_data_reference_name(
-        batch_identifiers: PartitionDefinitionSubset,
+        batch_identifiers: BatchIdentifiersSubset,
     ) -> str:
         if batch_identifiers is None:
-            batch_identifiers = PartitionDefinitionSubset({})
+            batch_identifiers = BatchIdentifiersSubset({})
         data_reference_name = DEFAULT_DELIMITER.join(
             [str(value) for value in batch_identifiers.values()]
         )
         return data_reference_name
 
+    @staticmethod
+    def _validate_runtime_parameters(runtime_parameters: Union[dict, type(None)]):
+        if not isinstance(runtime_parameters, dict):
+            raise TypeError(
+                f"""The type of runtime_parameters must be a dict object. The type given is
+        "{str(type(runtime_parameters))}", which is illegal.
+                        """
+            )
+        keys_present = [
+            key
+            for key, val in runtime_parameters.items()
+            if val is not None and key in ["batch_data", "query", "path"]
+        ]
+        if len(keys_present) != 1:
+            raise ge_exceptions.InvalidBatchRequestError(
+                "The runtime_parameters dict must have one (and only one) of the following keys: 'batch_data', "
+                "'query', 'path'."
+            )
+
     def _validate_batch_request(self, batch_request: BatchRequestBase):
         super()._validate_batch_request(batch_request=batch_request)
 
-        # Insure that batch_data and batch_request satisfy the "if and only if" condition.
+        runtime_parameters = batch_request.runtime_parameters
+        batch_identifiers = batch_request.batch_identifiers
         if not (
-            (
-                batch_request.batch_data is None
-                and (
-                    batch_request.partition_request is None
-                    or not batch_request.partition_request.get("batch_identifiers")
-                )
-            )
-            or (
-                batch_request.batch_data is not None
-                and batch_request.partition_request
-                and batch_request.partition_request.get("batch_identifiers")
-            )
+            (not runtime_parameters and not batch_identifiers)
+            or (runtime_parameters and batch_identifiers)
         ):
             raise ge_exceptions.DataConnectorError(
-                f"""RuntimeDataConnector "{self.name}" requires batch_data and partition_request to be both present or
+                f"""RuntimeDataConnector "{self.name}" requires runtime_parameters and batch_identifiers to be both 
+                present and non-empty or
                 both absent in the batch_request parameter.
                 """
             )
+        if runtime_parameters:
+            self._validate_runtime_parameters(runtime_parameters=runtime_parameters)
```

Comment:
I realize that in cases where the `path` `runtime_parameter` has more than one file in the directory, we cannot expect the user to specify the `batch_identifier` for every file... this may require a bit more thought. I actually have some free time tomorrow afternoon. I'll try to put a proposed usecase together.

Reply:
Oh. I was thinking `path` would be for a file, not a directory, which means there should also be a check for this (if this is indeed how we want `path` to work.